src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -241,37 +195,85 @@ export class Dropdown extends Component<DropdownProps> {
         id={id}
         {...passProps}
       >
-        <HtmlLabel
+        <LabelText
           id={labelId}
-          {...labelProps}
-          className={dropdownClassNames.label}
+          labelMode={labelMode}
+          as="label"
+          optionalText={optionalText}
         >
-          {labelMode === 'hidden' ? (
-            <VisuallyHidden>{labelText}</VisuallyHidden>
-          ) : (
-            <Paragraph {...labelTextProps}>{labelText}</Paragraph>
-          )}
-        </HtmlLabel>
+          {labelText}
+        </LabelText>
         <ListboxInput {...listboxInputProps}>
-          <ListboxButton {...passDropdownButtonProps}>
+          <ListboxButton
+            {...dropdownButtonProps}
+            id={buttonId}
+            className={classnames(
+              dropdownClassNames.button,
+              dropdownButtonProps.className,
+              {
+                [dropdownClassNames.disabled]: !!disabled,
+              },
+            )}
+            {...buttonAriaLabelledByOverride}
+          >
             {listboxDisplayValue}
           </ListboxButton>
           {!!ListboxPopoverComponentReplace ? (
             <ListboxPopoverComponentReplace {...passDropdownPopoverProps}>
-              {this.dropdownItems(children, passDropdownItemProps)}
+              {children}
             </ListboxPopoverComponentReplace>
           ) : (
             <ListboxPopover
               position={positionMatchWidth}
               {...passDropdownPopoverProps}
             >
-              <ListboxList>
-                {this.dropdownItems(children, passDropdownItemProps)}
-              </ListboxList>
+              <ListboxList>{children}</ListboxList>
             </ListboxPopover>
           )}
         </ListboxInput>
       </HtmlSpan>
     );
   }
 }
+
+const StyledDropdown = styled(
+  ({
+    tokens,
+    id: propId,
+    ...passProps
+  }: DropdownProps & InternalTokensProp) => (
+    <AutoId id={propId}>
+      {(id) => <BaseDropdown id={id} {...passProps} />}
+    </AutoId>
+  ),
+)`
+  ${(props) => baseStyles(props)}
+`;
+
+interface NewListboxPopoverProps extends BaseListboxPopoverProps, TokensProp {}
+
+const StyledListboxPopover = styled(
+  ({ tokens, children, ...passProps }: NewListboxPopoverProps) => (
+    <ListboxPopover position={positionMatchWidth} {...passProps}>
+      <ListboxList>{children}</ListboxList>
+    </ListboxPopover>
+  ),
+)`
+  ${(props) => listboxPopoverStyles(props.theme)}
+`;
+
+/**
+ * <i class="semantics" />
+ * Use for selectable dropdown with items.
+ */
+export class Dropdown extends Component<DropdownProps> {
+  render() {
+    const props = withSuomifiDefaultProps(this.props);
+    return (
+      <StyledDropdown
+        {...props}
+        listboxPopoverComponent={StyledListboxPopover}
```

Comment:
This abstraction is probably no longer necessary since we only support the version with suomifi styles? So we could just add the styles in the same place with StyledDropdown?

Reply:
Seems to work that way at a glance. Will change this 👍